cloud-init.pl

```diff
@@ -84,6 +84,25 @@ sub apply_user_data {
       system("sh -c \"$runcmd\"");
     }
   }
+
+  if (defined($data->{write_files})) {
+    foreach my $item (@{ $data->{write_files} }) {
+      open my $fh, ">", $item->{path};
+      printf $fh "%s", $item->{content};
+      if (defined($item->{permissions})) {
+        my $perms = oct($item->{permissions});
```

Comment:
To handle both dirs and files, a future improvement could be to check for a `umask` key here and apply that before the `mkpath` and `open` calls, I wouldn't make that a prerequisite for this.